Automatic pulling ExtraFileMaps without explicit mapping.

kwanghoon-meta opened the pull request:
Summary:
It comes handy when we are trying to load a module with ExtraFileMaps after JITing since [*BytecodeDeserializer*](https://www.internalfb.com/code/aros/[144773a900f80d6bdb2cbc35f817532200b545be]/xros/third-party/caffe2/caffe2/torch/csrc/jit/mobile/import.cpp?lines=196%2C389%2C408%2C569%2C611) loops over extra maps entries that is given from end-user APIs.

We can simply uses [getAllRecord](https://www.internalfb.com/code/fbsource/[c8b0b670f4782345c954a292dc20f562dd95e263]/xplat/caffe2/caffe2/serialize/inline_container.cc?lines=31%2C67%2C72%2C79%2C86%2C92%2C174%2C178%2C222%2C243%2C275%2C283%2C304%2C322) so that we can easily pull all extra files embedded without explicitly mapping with high level [API ](https://www.internalfb.com/code/fbsource/fbcode/caffe2/torch/csrc/jit/mobile/import.cpp?lines=548-563)

Note :
- Flatbuffer by nature already extracts all embedded files without explicit mapping. I'm simply adding extra unit test for future refence if someone also interested in knowing later on

- only lite interpreter needs a change as diff does

Test Plan:
buck2 run //caffe2/test/cpp/jit:jit -- --gtest_filter=LiteInterpreterTest.ExtraFiles

buck2 run //caffe2/test/cpp/jit:jit -- --gtest_filter=FlatbufferTest.ExtraFiles

Differential Revision: D45170126

davidberard98 reviewed (approved):
nit - address linter issues

otherwise LGTM!

kwanghoon-meta committed:
Automatic pulling ExtraFileMaps without explicit mapping. (#99747)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/99747

It comes handy when we are trying to load a module with ExtraFileMaps after JITing since [*BytecodeDeserializer*](https://www.internalfb.com/code/aros/[144773a900f80d6bdb2cbc35f817532200b545be]/xros/third-party/caffe2/caffe2/torch/csrc/jit/mobile/import.cpp?lines=196%2C389%2C408%2C569%2C611) loops over extra maps entries that is given from end-user APIs.

We can simply uses [getAllRecord](https://www.internalfb.com/code/fbsource/[c8b0b670f4782345c954a292dc20f562dd95e263]/xplat/caffe2/caffe2/serialize/inline_container.cc?lines=31%2C67%2C72%2C79%2C86%2C92%2C174%2C178%2C222%2C243%2C275%2C283%2C304%2C322) so that we can easily pull all extra files embedded without explicitly mapping with high level [API ](https://www.internalfb.com/code/fbsource/fbcode/caffe2/torch/csrc/jit/mobile/import.cpp?lines=548-563)

Note :
- Flatbuffer by nature already extracts all embedded files without explicit mapping. I'm simply adding extra unit test for future refence if someone also interested in knowing later on

- only lite interpreter needs a change as diff does

Test Plan:
buck2 run //caffe2/test/cpp/jit:jit -- --gtest_filter=LiteInterpreterTest.ExtraFiles

buck2 run //caffe2/test/cpp/jit:jit -- --gtest_filter=FlatbufferTest.ExtraFiles

Reviewed By: davidberard98

Differential Revision: D45170126

fbshipit-source-id: 57c8a43cd64931f5dcca4b36d306b23769d4ce5b
